lib/routes/password.js

```diff
@@ -204,22 +204,21 @@ module.exports = function (
         }
 
         function createSessionToken() {
-          if (sessionTokenId) {
-            var sessionTokenOptions = {
-              uid: account.uid,
-              email: account.email,
-              emailCode: account.emailCode,
-              emailVerified: account.emailVerified,
-              verifierSetAt: account.verifierSetAt
-            }
-
-            return db.createSessionToken(sessionTokenOptions, request.headers['user-agent'])
-              .then(
-                function (result) {
-                  sessionToken = result
-                }
-              )
+          // Create a sessionToken with the verification status of the current session
+          var sessionTokenOptions = {
+            uid: account.uid,
+            email: account.email,
+            emailCode: account.emailCode,
+            emailVerified: account.emailVerified,
+            verifierSetAt: account.verifierSetAt
```

Comment:
Despite the comment about using the verification status of the current session, this doesn't appear to take any state from the provided sessionToken.  Also, the `sessionToken` argument is still optional, so perhaps we should maintain the existing logic of only creating one if one was provided in the request.